### PR TITLE
.Net: Update classes so namespaces are consistent and match project location

### PR DESCRIPTION
**Pull request by markwallace-microsoft**
### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/Connectors/Connectors.HuggingFace/Core/TextEmbeddingRequest.cs [hunk 1]
```diff
@@ -3,7 +3,7 @@
 using System.Collections.Generic;
 using System.Text.Json.Serialization;
 
-namespace Microsoft.SemanticKernel.Connectors.HuggingFace;
+namespace Microsoft.SemanticKernel.Connectors.HuggingFace.Core;
```
Comment (dmytrostruk) [inline dotnet/src/Connectors/Connectors.HuggingFace/Core/TextEmbeddingRequest.cs:6]: This class is public, and for public classes we want to keep namespaces easy and short. That's why for public classes the rule to "align namespace to the project location" doesn't apply. But in general, if this class is used as HTTP request schema, then I think it should be internal.

File: dotnet/src/Connectors/Connectors.HuggingFace/Core/TextEmbeddingResponse.cs [hunk 2]
```diff
@@ -3,7 +3,7 @@
 using System;
 using System.Collections.Generic;
 
-namespace Microsoft.SemanticKernel.Connectors.HuggingFace;
+namespace Microsoft.SemanticKernel.Connectors.HuggingFace.Core;
```
Comment (dmytrostruk) [inline dotnet/src/Connectors/Connectors.HuggingFace/Core/TextEmbeddingResponse.cs:6]: Same comment as above.

File: dotnet/src/Connectors/Connectors.HuggingFace/Core/TextEmbeddingRequest.cs (hunk 1 above)
Comment (markwallace-microsoft) [reply dotnet/src/Connectors/Connectors.HuggingFace/Core/TextEmbeddingRequest.cs:6]: The request/response classes for text generation are internal as is the `HuggingFaceClient` so I'll make these two internal also.

**Review by dmytrostruk — APPROVED** [review]

**Comment by RogerBarreto** [comment]
Thanks for:
- Splitting the ServiceCollection in it's own class, the other connectors are doing in the same, this is a good move.
- Making internal the missing request/response `Models`, missed that.

REgarding the `namespace` change I don't fully agree, as this defeats the purpose of having specific modality `namespaces` where classes that could be in it are all mixed in `Core`.

**Review by RogerBarreto — APPROVED** [review]

**Pull request merged**